spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -1145,45 +1161,62 @@ def write_units(
                     unit_col_args.update(table=nwbfile.electrodes)
                 nwbfile.add_unit_column(**unit_col_args)
 
+            if write_waveforms:
```

Comment:
This requires `pynwb>=1.4.0`

Add this in the imports: 
```
from distutils.version import StrictVersion
```

And then this to the check:
```suggestion
            if write_waveforms and StrictVersion(pynwb.__version__) >= '1.4.0':
```

Reply:
So, the main reason for your errors below is the actual version of pynwb with the waveform addition is `v1.4.0.post.dev8`, which I believe is automatically assigned by their version tracking package. StrictVersion throws an error when applied to this since it doesn't think it's a proper version number.